packages/hardhat-chai-matchers/src/bigNumber.ts

```diff
@@ -0,0 +1,329 @@
+// TODO: consider adding support for the following Chai.Assert properties.
+// Might any of these come "for free" due to supporting other methods? At the
+// very least get them tested.
+//   isAbove
+//   isAtLeast
+//   isBelow
+//   isAtMost
+//   isAt
+//   isNumber
+//   isNotNumber
+//   closeTo
+//   approximately
```

Comment:
I think we can drop this comment. Most of these methods work. `isNumber` doesn't (that is, `expect(1n).to.be.a('number')` throws), but I think that's fine.

Reply:
Addressed in https://github.com/NomicFoundation/hardhat/pull/2428/commits/1dd96ce8a13a5676f42ca17111989c6ba3e02be8. I didn't completely drop it, because some of them actually don't work. But I revised it to remove the ones that do work, and to remove `isNumber` and `isNotNumber`.

Reply:
What, really? I'm pretty sure I tried `isAbove`, `isAtLeast`, `isBelow` and `isAtMost`. Maybe I did something wrong, but I'm fairly confident that they were working. I missed `isAt` though.

Reply:
Just remove this comment. It's for assertion types that are not `expect`, and everyone out there uses `expect`. So just leave it undone until someone complains about it.

Reply:
Addressed in https://github.com/NomicFoundation/hardhat/pull/2428/commits/2b0fc7464ee7fbd03dee901630b82aa9ce791963